zaza/openstack/charm_tests/ceph/tests.py

```diff
@@ -690,28 +702,162 @@ def expected_apps(self):
     def multisite(self):
         """Determine whether deployment is multi-site."""
         try:
-            zaza_model.get_application('slave-ceph-radosgw')
+            zaza_model.get_application(self.secondary_rgw_app)
             return True
         except KeyError:
             return False
 
+    def get_rgwadmin_cmd_skeleton(self, unit_name:str):
+        '''Get radosgw-admin cmd skeleton with rgw.hostname populated key'''
+        app_name = unit_name.split('/')[0]
+        juju_units = zaza_model.get_units(app_name)
+        unit_hostnames = generic_utils.get_unit_hostnames(juju_units)
+        hostname = unit_hostnames[unit_name]
+        return 'radosgw-admin --id=rgw.{} '.format(hostname)
+
     @tenacity.retry(wait=tenacity.wait_exponential(multiplier=10, max=300),
                     reraise=True, stop=tenacity.stop_after_attempt(10),
                     retry=tenacity.retry_if_exception_type(AssertionError))
-    def wait_for_sync(self, application):
+    def wait_for_sync(self, application, isPrimary=False):
         """Wait for slave to secondary to show it is in sync."""
```

Comment:
It looks like this function is more "Checking if the applications are in sync, rather than waiting for that state?

Reply:
Your observation is correct, by itself it only checks whether site is in sync, the last line assertion and the tenacity decorator make it wait for sync. Since it "effectively" waits and can't be used to "just check" the sync status, I kept the original name of the method.